components/sync15/src/request.rs

```diff
@@ -129,6 +137,7 @@ impl Deref for InfoCollections {
     }
 }
 
+// A struct to upload the result of a batch request
```

Comment:
Documenting this is good since it is public, but make this a [doc comment](https://doc.rust-lang.org/rust-by-example/meta/doc.html). 

"A struct" is redundant. Consider:

`/// The result of a batch upload, containing results for successful and failed updates.`